### ISSUE TITLE
Datepicker: change event is triggered with `mask=true` when leaving input without changes

### ISSUE THREAD

**Issue by SirError** [primefaces/primefaces#9259]
### Describe the bug

The first time I pick any date the ajax event is called twice, this can be checked through F12
the first ajax sends an empty date, and the second the value picked.
```
<p:datePicker id="value"
		      value="#{testView.date}"
		      mask="true"
		      pattern="dd/MM/yyyy"
		      showIcon="true"
		      showOnFocus="false"> 
	<p:ajax event="change" update="date"/>		 
</p:datePicker>	
```

### Reproducer

[primefaces-test-master.zip](https://github.com/primefaces/primefaces/files/9700633/primefaces-test-master.zip)

### Expected behavior

The event should be called just once with the correct value.

### PrimeFaces edition

Community

### PrimeFaces version

12.0.0.RC3

### JSF implementation

Mojarra

### JSF version

2

### Java version

2.2.20

**Comment by tandraschko** [comment]
thats related to the mask and #7176

@melloware interessted to check it?

**Comment by melloware** [comment]
I can take a look its most likely the mask is triggering and onChange event

**Comment by SirError** [comment]
Hum, seems like now the change event is not working when I type the date

**Comment by melloware** [comment]
Can you try it without the mask.  My guess is the change event never fired when you typed the date?

**Comment by melloware** [comment]
All I did was turn off the MASK onblur but nothing else of the component changed.

**Comment by SirError** [comment]
Yeah, without mask it works
but shouldn't it work with mask as well?

**Comment by melloware** [comment]
it does if I reinstate the change I just made and then you will get double AJAX requests.  So you have to determine which you want more.

**Comment by melloware** [comment]
The InputMask doesn't allow me to get at the underlying events so I have to brute force removing the "blur" if you are using the mask.  Let me try soemthing else.

**Comment by SirError** [comment]
The mask is important to my use case, so I prefer double event over no event =x

**Comment by melloware** [comment]
OK reverted.  Because InputMask adds so many events under the covers I recommend if you really need this use case just stop using mask for now.

**Comment by jan666** [comment]
This does not work either:

```
<p:datePicker id="dp" value="#{page.testDateInput}" mask="true" navigator="true" pattern="dd.MM.yyyy" hideOnDateTimeSelect="true" yearNavigator="true" monthNavigator="true" required="true" requiredMessage="required">
		<p:ajax event="change"	update="@form"	/>
		<p:ajax event="dateSelect"	update="@form"	/>
</p:datePicker>
```

the value is not submitted (and because its required I get the "reuqired" error):

```
javax.faces.partial.ajax: 
true
javax.faces.source: 
j_id_3c:j_id_3e:dp
javax.faces.partial.execute: 
j_id_3c:j_id_3e:dp
javax.faces.partial.render: 
j_id_3c:j_id_3e
javax.faces.behavior.event: 
dateSelect
javax.faces.partial.event: 
dateSelect
j_id_3c:j_id_3e:dp_input: 
j_id_3c:j_id_3e:j_id_3i_focus: 
j_id_3c:j_id_3e:j_id_3i_input: 
1
j_id_3c:j_id_3e:j_id_3l: 
j_id_3c:j_id_3e_SUBMIT: 
1
```

```
javax.faces.partial.ajax: 
true
javax.faces.source: 
j_id_3c:j_id_3e:dp
javax.faces.partial.execute: 
j_id_3c:j_id_3e:dp
javax.faces.partial.render: 
j_id_3c:j_id_3e
javax.faces.behavior.event: 
change
javax.faces.partial.event: 
change
j_id_3c:j_id_3e:dp_input: 
j_id_3c:j_id_3e:j_id_3i_focus: 
j_id_3c:j_id_3e:j_id_3i_input: 
1
j_id_3c:j_id_3e:j_id_3l: 
j_id_3c:j_id_3e_SUBMIT: 
1
```

Works without mask/pattern:

```
<p:datePicker id="dp" value="#{page.testDateInput}" navigator="true" hideOnDateTimeSelect="true" yearNavigator="true" monthNavigator="true" required="true" requiredMessage="required">
		<p:ajax event="change"	update="@form"	/>
		<p:ajax event="dateSelect"	update="@form"	/>
</p:datePicker>
```

```
javax.faces.partial.ajax: 
true
javax.faces.source: 
j_id_3c:j_id_3e:dp
javax.faces.partial.execute: 
j_id_3c:j_id_3e:dp
javax.faces.partial.render: 
j_id_3c:j_id_3e
javax.faces.behavior.event: 
dateSelect
javax.faces.partial.event: 
dateSelect
j_id_3c:j_id_3e:dp_input: 
31.12.2022
j_id_3c:j_id_3e:j_id_3i_focus: 
j_id_3c:j_id_3e:j_id_3i_input: 
1
j_id_3c:j_id_3e:j_id_3l: 
j_id_3c:j_id_3e_SUBMIT: 
1
```

```
javax.faces.partial.ajax: 
true
javax.faces.source: 
j_id_3c:j_id_3e:dp
javax.faces.partial.execute: 
j_id_3c:j_id_3e:dp
javax.faces.partial.render: 
j_id_3c:j_id_3e
javax.faces.behavior.event: 
change
javax.faces.partial.event: 
change
j_id_3c:j_id_3e:dp_input: 
31.12.2022
j_id_3c:j_id_3e:j_id_3i_focus: 
j_id_3c:j_id_3e:j_id_3i_input: 
1
j_id_3c:j_id_3e:j_id_3l: 
j_id_3c:j_id_3e_SUBMIT: 
1
```

**Comment by melloware** [comment]
Basically from your data I am gathering that the `change` `dateSelect` AJAX event ordering is different?  Other than that it looks identical to me?

**Comment by jan666** [comment]
The value is not submitted when using the mask. It's just empty. Thats the problem. Not at the change, not at the dateSelect event (I clicked on a date in the calender popup)

**Comment by melloware** [comment]
Submittted patch to InputMask to help me fix this: https://github.com/RobinHerbots/Inputmask/issues/2672

**Comment by jan666** [comment]
I tried your branch (13.0.0-SNAPSHOT) and it works again (I guess there is no workaround for 12?)

**Comment by melloware** [comment]
@jan666 here is a reproducer I patched for 12.0.0: 
[pf-9259-patched.zip](https://github.com/primefaces/primefaces/files/10176310/pf-9259-patched.zip)

Just look in `\src\main\webapp\resources\primefaces` by including those two patches JS files at the right location they override the ones found in 12.0.0 because the ones in your WAR take precedence.

**Comment by jan666** [comment]
Works great. Thanks!
